cyclonedx-lib/sign_src/TemurinSignSBOM.java

```diff
@@ -0,0 +1,191 @@
+            try {
+                privateKeyFileInputStream.read(privateKeyData);
+            } finally {
+                privateKeyFileInputStream.close();
+            }
+            String privateKeyString = new String(privateKeyData, "UTF-8");
+            KeyPair sampleKey = PEMDecoder.getKeyPair(privateKeyData);
+
+            // Sign the JSON data
+            String signedData = new JSONObjectWriter(JSONParser.parse(sbomDataToSign))
+                    .setSignature(new JSONAsymKeySigner(sampleKey.getPrivate()))
+                    .serializeToString(JSONOutputFormats.PRETTY_PRINT);
+
+            JsonParser parser = new JsonParser();
+            Bom signedBom = parser.parse(new StringReader(signedData));
+            return signedBom;
+        } catch (IOException | GeneralSecurityException | org.cyclonedx.exception.ParseException e) {
+            // Log the exception with the logger
+            LOGGER.severe("An error occurred while signing the SBOM: " + e.getMessage());
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    static String generateBomJson(final Bom bom) {
+        BomJsonGenerator bomGen = BomGeneratorFactory.createJson(CycloneDxSchema.Version.VERSION_14, bom);
+        String json = bomGen.toJsonString();
+        return json;
+    }
+
+    static void writeJSONfile(final Bom bom, final String fileName) {
+        // Creates testJson.json file
+        FileWriter file;
+        String json = generateBomJson(bom);
+        try {
+            file = new FileWriter(fileName);
+            file.write(json);
+            file.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    static Bom readJSONfile(final String fileName) {                                 // Returns parse bom
+        Bom bom = null;
+        try {
+            FileReader reader = new FileReader(fileName);
+            JsonParser parser = new JsonParser();
+            bom = parser.parse(reader);
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            return bom;
+        }
+    }
+
+    static boolean verifySignature(final String jsonFile, final String publicKeyFile) {
+        try {
+            // Read the JSON file to be verified
+            Bom bom = readJSONfile(jsonFile);
+            String signedSbomData = generateBomJson(bom);
+
+            // Parse JSON
+            JSONObjectReader reader = JSONParser.parse(signedSbomData);
+
+            // Load public key from file
+            PublicKey publicKey = loadPublicKeyFromFile(publicKeyFile);
+
+            // Verify signature using the loaded public key
+            JSONSignatureDecoder signature = reader.getSignature(new JSONCryptoHelper.Options());
+            signature.verify(new JSONAsymKeyVerifier(publicKey));
+            return true;
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Use of printStackTrace() is not recommended generally in non-debug code.
Put a suitable System.out.println("Exception verifying json signature: "+e); message here